pandas/core/nanops.py

```diff
@@ -388,13 +391,14 @@ def _na_for_min_count(values, axis):
     if values.ndim == 1:
         return fill_value
     else:
+        assert axis is not None  # assertion to make mypy happy
         result_shape = values.shape[:axis] + values.shape[axis + 1 :]
         result = np.empty(result_shape, dtype=values.dtype)
         result.fill(fill_value)
         return result
 
 
-def nanany(values, axis=None, skipna=True, mask=None):
+def nanany(values, axis=None, skipna: bool = True, mask=None):
```

Comment:
Not sure if applicable here but you could use the `Axis` definition from pandas._typing, if this can accept both integers and "index"/"columns"

Annotated `bool` return value for this and below would be great too